core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -194,15 +192,34 @@ public static String decodeString(String encoded) {
    * @return UTF-8 encoded string representing the namespace, suitable for use as a URL parameter
    */
   public static String encodeNamespace(Namespace ns) {
-    Preconditions.checkArgument(ns != null, "Invalid namespace: null");
-    String[] levels = ns.levels();
+    return encodeNamespace(ns, NAMESPACE_ESCAPED_SEPARATOR);
+  }
+
+  /**
+   * Returns a String representation of a namespace that is suitable for use in a URL / URI.
+   *
+   * <p>This function needs to be called when a namespace is used as a path variable (or query
+   * parameter etc.), to format the namespace per the spec.
+   *
+   * <p>{@link RESTUtil#decodeNamespace(String, String)} should be used to parse the namespace from
+   * a URL parameter.
+   *
+   * @param namespace namespace to encode
+   * @param separator The namespace separator to use for encoding
+   * @return UTF-8 encoded string representing the namespace, suitable for use as a URL parameter
```

Comment:
What's a "URL parameter"? A URI path element? A URI query part?

Reply:
this is just copy-pasted javadoc from the overloaded method. no new stuff is being added, so the wording is exactly the same

Reply:
Clarification would help nonetheless.
The code/javadoc is being added here - so I think it's justified to clarify things.